apply_DDT.py

```diff
@@ -0,0 +1,594 @@
+#==============================================================================
+# apply_DDT.py ----------------------------------------------------------------
+#------------------------------------------------------------------------------
+# Author(s): Brendan Regnery, Sara Nabili -------------------------------------
+#------------------------------------------------------------------------------
+# Applies a DDT to a trained BDT model ----------------------------------------
+#    (Designed Decorrelated Tagger, https://arxiv.org/pdf/1603.00027.pdf) -----
+#------------------------------------------------------------------------------
+
+import os, os.path as osp, glob, pickle, logging, warnings, json, math, re
+from time import strftime
+from collections import OrderedDict
+from common import create_DDT_map_dict, calculate_varDDT
+import json
+
+import numpy as np
+import matplotlib
+matplotlib.use('Agg') # prevents opening displays (fast), must use before pyplot
+import matplotlib.pyplot as plt
+from sklearn.metrics import confusion_matrix, roc_curve, roc_auc_score, auc
+import pandas as pd
+import xgboost as xgb
+from scipy.ndimage import gaussian_filter
+import mplhep as hep
+hep.style.use("CMS") # CMS plot style
+
+#disable tex formating while having problems
+#plt.rcParams['text.usetex'] = False
+
+np.random.seed(1001)
+
+from common import logger, DATADIR, Columns, time_and_log, imgcat, set_mpl_fontsize, columns_to_numpy
+
+#------------------------------------------------------------------------------
+# User input global variables (to document options during running) ------------
+#------------------------------------------------------------------------------
+
+training_features = [
+    'girth', 'ptd', 'axismajor', 'axisminor',
+    'ecfm2b1', 'ecfd2b1', 'ecfc2b1', 'ecfn2b2', 'metdphi',
+    'ak15_chad_ef', 'ak15_nhad_ef', 'ak15_elect_ef', 'ak15_muon_ef', 'ak15_photon_ef', 
+    ]
+
+bkg_data_files = 'data/bkg_20240515/Summer20UL18/*.npz'
```

Comment:
should this (and the various other variables below) be command line arguments?

Reply:
This I wasn't sure about, in my past analysis group we liked to keep the arguments listed at the top of the code like this so that during pre-approval we could issue a code tag that corresponds to a certain AN/PAS on git and immediately see all the commands that had been run to create it. But I know Thomas's style is quite different. I can make them command line arguments if you think it is needed.

Reply:
I personally tend toward the approach of making everything configurable and then tagging specific configurations (e.g. by putting the commands in a shell script).

Reply:
Sure, makes sense then. I always have a bit of trouble deciding what should or shouldn't be a command line option, which variables would you like me to make options?

Reply:
file locations, lumi value, cut values. maybe which plots to make. the list of training features isn't changing at this point, so that can be left alone.